### PR TITLE
Feature/azure support

### PR DESCRIPTION
**Pull request by geclos**
https://www.loom.com/share/5da46599b7484cb1a5ceccb80f107088?sid=3907bd8c-a999-4107-b74f-e60543ddb530

### CODE REVIEW

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx [hunk 1]
```diff
@@ -24,23 +24,25 @@ import {
...
 
 import { Header } from './Header'
 import Playground from './Playground'
 
 export const DocumentEditorContext = createContext<
-  { runAction: Function; streamTextAction: Function } | undefined
+  | {
+      streamTextAction: StreamTextOutputAction
+    }
+  | undefined
```
Comment (csansoon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx:38]: Are these changes on top of #97 ? Feels like it.

File: packages/core/src/services/ai/index.ts [hunk 2]
```diff
@@ -31,27 +32,37 @@ type FinishCallbackEvent = {
...
 type FinishCallback = (event: FinishCallbackEvent) => void
 
+export type Config = {
+  provider: string
+  model: string
+  azure?: { resourceName: string }
+} & Record<string, unknown>
+export type PartialConfig = Omit<Config, 'provider' & 'model'>
+
+const GROR_API_URL = 'https://api.groq.com/openai/v1'
```
Comment (andresgutgon) [inline packages/core/src/services/ai/index.ts:44]: typo.

Comment (csansoon) [reply packages/core/src/services/ai/index.ts:44]: Move this to a constants.ts instead

File: packages/core/src/services/commits/runDocumentAtCommit.ts [hunk 3]
```diff
@@ -183,11 +171,24 @@ async function doSomeCommonOperations({
...
+
+async function findApiKey({
+  scope,
+  name,
+}: {
+  scope: ProviderApiKeysRepository
+  name: string
+}) {
+  const apiKeyResult = await scope.findByName(name)
+  if (apiKeyResult.error) throw apiKeyResult.error
```
Comment (andresgutgon) [inline packages/core/src/services/commits/runDocumentAtCommit.ts:191]: Just return the result no the `.error`. Also why `value!`? It can be `null`?

**Review by csansoon — APPROVED** [review]

File: packages/core/src/services/commits/runDocumentAtCommit.ts (hunk 3 above)
Comment (geclos) [reply packages/core/src/services/commits/runDocumentAtCommit.ts:191]: nah this is for internal use no real need to return result, i want this to throw unless it works

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx (hunk 1 above)
Comment (geclos) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx:38]: yup, talked offline. Gonna merge this, feel free to close #97

**Review by andresgutgon — APPROVED** [review]

**Pull request merged**